js/packages/openinference-instrumentation-llama-index/src/utils.ts

```diff
@@ -161,25 +228,90 @@ export function patchRetrieveMethod(
   };
 }
 
-function documentAttributes(
-  output: llamaindex.NodeWithScore<llamaindex.Metadata>[],
+export function patchQueryEmbedding(
+  original: QueryEmbeddingMethod,
+  tracer: Tracer,
 ) {
-  const docs: Attributes = {};
-  output.forEach((document, index) => {
-    const { node, score } = document;
+  return function patchedQueryEmbedding(
+    this: unknown,
+    ...args: Parameters<QueryEmbeddingMethod>
+  ) {
+    const span = tracer.startSpan(`embedding`, {
+      kind: SpanKind.INTERNAL,
+      attributes: {
+        [SemanticConventions.OPENINFERENCE_SPAN_KIND]:
+          OpenInferenceSpanKind.EMBEDDING,
+      },
+    });
 
-    if (node instanceof TextNode) {
-      const nodeId = node.id_;
-      const nodeText = node.getContent();
-      const nodeMetadata = node.metadata;
+    const execContext = getExecContext(span);
 
-      const prefix = `${SemanticConventions.RETRIEVAL_DOCUMENTS}.${index}`;
-      docs[`${prefix}.${SemanticConventions.DOCUMENT_ID}`] = nodeId;
-      docs[`${prefix}.${SemanticConventions.DOCUMENT_SCORE}`] = score;
-      docs[`${prefix}.${SemanticConventions.DOCUMENT_CONTENT}`] = nodeText;
-      docs[`${prefix}.${SemanticConventions.DOCUMENT_METADATA}`] =
-        safelyJSONStringify(nodeMetadata) ?? undefined;
-    }
-  });
-  return docs;
+    const execPromise = safeExecuteInTheMiddle<
+      ReturnType<QueryEmbeddingMethod>
+    >(
+      () => {
+        return context.with(execContext, () => {
+          return original.apply(this, args);
+        });
+      },
+      (error) => handleError(span, error),
+    );
+
+    // Model ID/name is a property found on the class and not in args
+    // Extract from class and set as attribute
+    span.setAttributes({
+      [SemanticConventions.EMBEDDING_MODEL_NAME]: getModelName(this),
+    });
```

Comment:
just curious i forgot to ask before is id really the same thing as a model name?

Reply:
Double checked and is actually not the same thing as model name as id seems to refer to some node id (https://github.com/run-llama/LlamaIndexTS/blob/c825a2f7436a55017f769ba2e7ac614a253709db/packages/core/src/schema/type.ts). Thanks for the catch!